DB/NEW_KT_DB/DataAccess/ObjectManager.py

```diff
@@ -58,47 +80,52 @@ def convert_object_name_to_management_table_name(object_name):
         return f'mng_{object_name}s'
 
 
-    def is_management_table_exist(table_name):
-        # check if table exists using single result query
-        return db_manager.execute_query_with_single_result(f'desc table {table_name}')
+    def is_management_table_exist(self, table_name):
+        # Check if table exists by querying the sqlite_master table
+        query = f"SELECT name FROM sqlite_master WHERE type='table' AND name='{table_name}'"
+        return self.db_manager.execute_query_with_single_result(query)
 
 
     # for outer use:
     def save_in_memory(self, object_name:str, metadata:Dict[str:Any], object_id:Optional[str] = None):
-       
+
         # insert object info into management table mng_{object_name}s
         # for exmple: object db_instance will be saved in table mng_db_instances
         table_name = self.convert_object_name_to_management_table_name(object_name)
 
         if not self.is_management_table_exist(table_name):
             self.create_management_table(table_name)
-        
+
         if object_id:
             self.insert_object_to_management_table_with_str_id(table_name, object_id, metadata)
-        
-        insert_object_to_management_table(table_name, object)
+
+        else:
+            self.insert_object_to_management_table(table_name, metadata)
 
     
-    def delete_from_memory(self,object_name, criteria='default', object_id:Optional[str] =  None):
-        
+    def delete_from_memory(self,object_name:str, criteria='default', object_id:Optional[str] =  None):
+
         # if criteria not sent- use PK for deletion
         if criteria == 'default':
-            criteria = f'{self.pk_column} = {self.pk_value}'
-        
+            if not object_id:
+                raise ValueError('must be or criteria or object id')
+            criteria = f'object_id = {object_id}'
+
         table_name = self.convert_object_name_to_management_table_name(object_name)
-        
-        delete_data_from_table(table_name, criteria)
+
+        self.delete_object_from_management_table(table_name, criteria)
 
 
     def update_in_memory(self, object_name, updates, criteria='default', object_id:Optional[str] =  None):
-        
+
         # if criteria not sent- use PK for deletion
         if criteria == 'default':
-            criteria = f'{self.pk_column} = {self.pk_value}'
+            if not object_id:
+                raise ValueError('must be or criteria or object id')
+            criteria = f'object_id = {object_id}'
```

Comment:
pk column name

Reply:
maybe pk_coulumn need to be class member in ObjectManager class?

Reply:
from my manager I send pk_column name and objectName as parameters to ObjectManager init, and then object name can convert to table name in init and not in every function.